src/main/java/com/wolox/training/providers/UserAuthProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.wolox.training.providers;
+
+import com.wolox.training.exceptions.InvalidAuthenticationException;
+import com.wolox.training.models.User;
+import com.wolox.training.repositories.UserRepository;
+import java.util.Collections;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserAuthProvider implements AuthenticationProvider {
+
+  @Autowired
+  private PasswordEncoder passwordEncoder;
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @Override
+  public Authentication authenticate(Authentication authentication) {
+    String username = authentication.getName();
+    String password = authentication.getCredentials().toString();
+    User user = userRepository.findFirstByUsername(username)
+        .orElseThrow(InvalidAuthenticationException::new);
+    if (!passwordEncoder.matches(password, user.getPassword())) {
+      throw new InvalidAuthenticationException();
+    }
+    return new UsernamePasswordAuthenticationToken(username, password, Collections.emptyList());
```

Comment:
`Collections.emptyList()` is not necessary, this param is when you are using roles